ZenPacks/zenoss/OpenStackInfrastructure/apiclients/session.py

```diff
@@ -330,16 +332,20 @@ def _detect_api_version(self):
             try:
                 data = json.loads(response_body)
                 version_url = {}
+                best_version = "v0.0"
                 for version in data['versions']['values']:
                     if version['status'] != 'stable':
                         continue
+                    if LooseVersion(version['id']) < LooseVersion(best_version):
+                        continue
+                    if LooseVersion(version['id']) >= LooseVersion("v4.0"):
+                        continue
+                    best_version = version['id']
```

Comment:
I started to do it that way, but we're checking to see if a key is present in a dictionary.  I was muddling through how to check if a key fragment is present in a dictionary, when I started to wonder what would happen if they had v3.7 _and_ v3.8.  The URL structure suggests this is unlikely, but we weren't expecting them to give us a v3.8 instead of a v3.0.  It's not paranoia if they really are out to get you.... 

It _is_ late in the game to introduce non-trivial changes, though.  Let me see if I can come up with a version that works, but changes the behavior less.